server/src/document/document.service.ts

```diff
@@ -151,13 +232,29 @@ export class DocumentService {
         )
       `);
 
+      // We can't determine whether the record id pertains to a Package
+      // or Artifact until we try to query either one.
+      // So if firstPackage is undefined, we then attempt to retrieve an
+      // Artifact
       if (!firstPackage) {
-        const errorMessage = `No document access.`;
-        console.log(`Client attempted to retrieve document ${pathSegment}, but no associated public, submitted packages were found.`);
-        throw new HttpException({
-          "code": "NO_DOCUMENT_ACCESS",
-          "detail": errorMessage,
-        }, HttpStatus.FORBIDDEN);
+        const { records: [firstArtifact] } = await this.crmService.get('dcp_artifactses', `
```

Comment:
Oh my good god, "dcp_artifactses"?! The naming just get better and better!

Reply:
Yuuuuuup. This has also happened before with some of the form entities, like `subjectsites` I think. 🙈